### DISCUSSION TITLE
Create a Kafka source with security and authentication params

### DISCUSSION THREAD

**Discussion by indyMccarthy** [risingwavelabs/risingwave#3409] [Q&A] [answered]
Hey RisingWave experts 👋

As it is not documented at the moment I think, I was wondering if your Kafka Connector supports authent and security config by any way while creating SOURCE.

When I create my consumer in Java it looks like that:
``` java
        Properties props = new Properties();
        props.put("bootstrap.servers", servers);
        props.put("sasl.mechanism", "SCRAM-SHA-256");
        props.put("security.protocol", "SASL_SSL");
        props.put("sasl.jaas.config", "org.apache.kafka.common.secu..............");
        props.put("key.deserializer", keyDeserializer);
        props.put("value.deserializer", valueDeserializer);
        props.put("auto.offset.reset", autoOffsetReset);
        props.put("group.id", groupId);
        return new KafkaConsumer<>(props);
```
I'm particularly interested by `sasl.mechanism`, `sasl.jaas.config` and `security.protocol` param.

Thanks !

**Comment by tabVersion** [comment, marked as answer]
Thank you for your question. Security is indeed an integral part of saas services nowadays, but the authentication of external services is not our focus in our current development plan.

In my opinion, risingwave should be deployed on the public cloud and we can access the service provider's sdk to complete the authentication. Or a private deployment in a trusted environment, where external services like kafka should be trusted and there is no need to use complex authentication.

**Reply by tabVersion** [reply]
now we are supporting security related params, https://github.com/risingwavelabs/risingwave/pull/5693 🤣

**Reply by indyMccarthy** [reply]
Great ! Thanks for notifying, I will test it 🚀
